corehq/apps/dhis2/api.py

```diff
@@ -0,0 +1,134 @@
+import json
+import logging
+import requests
+from corehq.apps.dhis2.dbaccessors import get_dhis2_connection
+
+
+logger = logging.getLogger('json_api_logger')
+
+
+class JsonApiError(Exception):
+    """
+    JsonApiError is raised for HTTP or socket errors.
+    """
+    pass
+
+
+def json_serializer(obj):
+    """
+    A JSON serializer that serializes dates and times
+    """
+    if hasattr(obj, 'isoformat'):
+        return obj.isoformat()
+
+
+def log_request(func):
+
+    def log(log_level, json_api_request, request_error, response_status, response_body, path, data=None, **params):
+        """
+        Unpack path and data from args and kwargs in order to log them separately
+        """
+        logger.log(log_level, {
+            'domain': json_api_request.domain_name if json_api_request.domain_name is not None else '[N/A]',
+            'request_method': func.__name__.upper(),
+            'request_url': json_api_request.server_url + path,
+            'request_headers': json.dumps(json_api_request.headers),
+            'request_params': json.dumps(params),
+            'request_body': '' if data is None else json.dumps(data, default=json_serializer),
```

Comment:
could do this: https://github.com/dimagi/commcare-hq/blob/420c9b5b0cb13731933f3e85c9c7a9e4b7478f68/corehq/apps/repeaters/repeater_generators.py#L153-L153